### PR TITLE
Enhance webhook docs

### PR DESCRIPTION
**Pull request by pixelbrackets**
Since I stumpled upon the webhooks (and slash command) and did not understand the basic concept at first, I wanted to share my insights with others. Personally I have the feeling that the docs were written for people who are used to the integrations in Slack and already know the basic concepts of hooks, payloads and so on. But for newbies like me it was a bit confusing. So I hope I can help to improve the situation.

PS: I did not find any documentation of how the JSON payload is supposed to look like. The docs currently only refer to Slack at several points. But there seems to be no page which describes all available fields in the JSON payload. I created a [feature request](https://mattermost.uservoice.com/forums/306457-general/suggestions/19787740-documentation-of-json-payload-format-for-integrati) for this as well.

### CODE REVIEW

**Comment by JeffSchering** [comment]
Thanks @pixelbrackets!

File: source/developer/slash-commands.rst [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
-- **Custom commands support auto-complete.** When you a create a custom command for your teammates, you have the option to fill in information about how auto-complete should work with that command. This gives your teammates quick and easy access to use your custom slash command
+- **The general concept is the same as for outgoing webhooks.** Read about :doc:`Outgoing Webhooks <../developer/webhooks-outgoing/>` first to understand the basic concept.
+- **Mattermost slash commands are Slack-compatible.** If you have used Slack's slash commands previously to interact with external applications, you can reuse those same applications with Mattermost. Mattermost will automatically translate Slack's proprietary JSON payload format.
+- **Custom commands support auto-complete.** When you a create a custom command for your teammates, you have the option to fill in information about how auto-complete should work with that command. This gives your teammates quick and easy access to use your custom slash command.
```
Comment (lindy65) [inline source/developer/slash-commands.rst]: ..."When you a create a..." > "When you create a"

File: source/developer/webhooks-outgoing.rst [hunk 2]
```diff
@@ -3,17 +3,19 @@
 Outgoing Webhooks
 =================
 
-Outgoing webhooks allow external applications, written in the programming language of your choice--to receive HTTP POST requests whenever a user posts to a certain channel, with a trigger word at the beginning of the message, or a combination of both. If the external application responds appropriately to the HTTP request, as response post can be made in the channel where the original post occurred.
+In addition to :doc:`Incoming Webhooks <../developer/webhooks-incoming/>` Mattermost also supports outgoing webhooks. This allows Mattermost to send a request to a webservice and process the response. The outgoing webhook is triggered whenever a user posts to a certain channel, with a trigger word at the beginning of a message, or a combination of both. If the application responds appropriately to the HTTP request, a response message can be posted in the channel where the original post occurred.
```
Comment (lindy65) [inline source/developer/webhooks-outgoing.rst]: "webservice" > "web service"

File: source/developer/webhooks-outgoing.rst [hunk 3]
```diff
@@ -3,17 +3,19 @@
...
-- **Mattermost outgoing webhooks support full markdown.** When an integration responds with a message to post, it will have access to a rich range of formatting unavailable in Slack that is made possible through :doc:`markdown support <../help/messaging/formatting-text>` in Mattermost, including headings, formatted fonts, tables, inline images and other options supported by Mattermost Markdown.
-- **Outgoing webhooks are only supported in public channels.** If you need something that works in a private channel, consider using a :doc:`Slash Command <slash-commands>`.
+- **The outgoing webhook is similar to the incoming webhook.** As described in the chapter :doc:`Incoming Webhooks <../developer/webhooks-incoming/>` the application may be written in the the programming language of your choice. It needs to provide a URL which reacts to the request send by your Mattermost instance, and may be able to send a HTTP POST in the required JSON format as a response.
```
Comment (lindy65) [inline source/developer/webhooks-outgoing.rst]: ..."written in the the..." > "written in the"

**Comment by pixelbrackets** [comment]
Hey, thanks for your fast response. I fixed the listed errors and other grammatical typos.

**Pull request merged**